cli/src/dsym/mod.rs

```diff
@@ -0,0 +1,211 @@
+            if let Some(uuid_end) = uuid_part.find(' ') {
+                let uuid = &uuid_part[..uuid_end];
+                // Uppercase for standard UUID format
+                return Ok(uuid.to_uppercase());
+            }
+        }
+    }
+
+    anyhow::bail!(
+        "Could not extract UUID from dSYM at {}. dwarfdump output: {}",
+        dsym_path.display(),
+        stdout
+    )
+}
+
+/// Zip a dSYM bundle into memory
+fn zip_dsym_bundle(dsym_path: &PathBuf) -> Result<Vec<u8>> {
+    use std::io::{Cursor, Write};
+    use std::fs::File;
+    use std::io::Read;
+    use walkdir::WalkDir;
+
+    let mut buffer = Cursor::new(Vec::new());
+    
+    {
+        let mut zip = zip::ZipWriter::new(&mut buffer);
+        let options = zip::write::SimpleFileOptions::default()
+            .compression_method(zip::CompressionMethod::Deflated);
+
+        let _dsym_name = dsym_path
+            .file_name()
+            .ok_or_else(|| anyhow!("Invalid dSYM path"))?
+            .to_string_lossy();
+
+        for entry in WalkDir::new(dsym_path) {
+            let entry = entry?;
+            let path = entry.path();
+            
+            // Create relative path within the zip
+            let relative_path = path.strip_prefix(dsym_path.parent().unwrap_or(dsym_path))?;
+            let zip_path = relative_path.to_string_lossy();
+
+            if path.is_file() {
+                zip.start_file(zip_path.to_string(), options)?;
+                let mut file = File::open(path)?;
+                let mut contents = Vec::new();
+                file.read_to_end(&mut contents)?;
+                zip.write_all(&contents)?;
+            } else if path.is_dir() && path != dsym_path.as_path() {
+                // Add directory entry (but not the root)
+                zip.add_directory(format!("{}/", zip_path), options)?;
+            }
+        }
+
+        zip.finish()?;
+    }
+
+    Ok(buffer.into_inner())
+}
+
+/// Find all dSYM bundles in a directory
+pub fn find_dsym_bundles(directory: &PathBuf) -> Result<Vec<PathBuf>> {
+    use walkdir::WalkDir;
+
+    let mut dsyms = Vec::new();
+
+    for entry in WalkDir::new(directory).follow_links(true) {
+        let entry = entry?;
+        let path = entry.path();
+
+        if path.is_dir() {
+            if let Some(ext) = path.extension() {
+                if ext == "dSYM" {
+                    dsyms.push(path.to_path_buf());
+                }
+            }
+        }
+    }
+
+    Ok(dsyms)
+}
+
+/// Info extracted from a dSYM's Info.plist
+#[derive(Debug, Clone, Default)]
+pub struct PlistInfo {
+    /// CFBundleIdentifier (e.g., com.example.app)
+    pub bundle_identifier: Option<String>,
+    /// CFBundleShortVersionString (e.g., 1.2.3)
+    pub short_version: Option<String>,
+    /// CFBundleVersion (e.g., 42)
+    pub bundle_version: Option<String>,
+}
+
+/// Extract version info from a dSYM bundle's Info.plist
+pub fn extract_plist_info(dsym_path: &PathBuf) -> Result<PlistInfo> {
```

Comment:
we need that for RN/Flutter as well
see [thread](https://posthog.slack.com/archives/C07AA937K9A/p1763463463959149)
can we make this method reusable?